torch_glow/src/CachingGraphRunner.h

```diff
@@ -20,18 +20,52 @@
 #include <torch/csrc/jit/custom_operator.h>
 #include <torch/csrc/jit/ir.h>
 
-#include "glow/ExecutionEngine/ExecutionEngine.h"
+#include "glow/Runtime/HostManager/HostManager.h"
 
 namespace glow {
 
 /// Responsible for maintaining a mapping from PyTorch subgraphs and their
-/// unique input types to compiled Glow Functions.
+/// unique input types to a compiled Glow Function.
 class CachingGraphRunner {
-  /// Glow ExecutionEngine.
-  glow::ExecutionEngine executionEngine_;
+  struct PerGlowGraphInfo {
+    std::vector<glow::Placeholder *> inputPlaceholders;
+    std::vector<glow::Placeholder *> outputPlaceholders;
+
+    /// Name of the Glow function maintained by HostManager for this subgraph.
+    std::string functionName;
+
+    /// The unique hash for the subgraph and input shapes represented by this
+    /// PerGlowGraphInfo.
+    size_t hash;
+
+    /// The PyTorch node containing the subgraph that this PerGlowGraphInfo
+    /// represents.
+    const torch::jit::Node *node;
+  };
+
+  std::unique_ptr<runtime::HostManager> hostManager_;
+  std::unordered_map<size_t, std::unique_ptr<PerGlowGraphInfo>>
```

Comment:
When do we free up `PerGlowGraphInfo` elements?

Reply:
This is a good question, right now we never do (except at shutdown).
There are two possibilities:
1) find a hook for when graph get removed from JIT and remove all associated Glow function then (this would be ideal). In fact I think we could do this by keep a smart ptr to this structure in the Glow fusion op. I will try this as a follow up.
2) create max cache size here and remove the lru when it's hit (not as ideal)

Reply:
Yeah, smart pointer should be good.

Reply:
Added a TODO for this